src/gui/src/dbDescriptors.cpp

```diff
@@ -1821,4 +1836,355 @@ bool DbItermAccessPointDescriptor::getAllObjects(SelectionSet& objects) const
+
+  auto* gui = Gui::get();
+
+  Properties props({{"Region Type", region->getRegionType().getString()}});
+  auto* parent = region->getParent();
+  if (parent != nullptr) {
+    props.push_back({"Parent", gui->makeSelected(parent)});
+  }
+
+  SelectionSet children;
+  for (auto* child : region->getChildren()) {
+    children.insert(gui->makeSelected(child));
+  }
+  if (!children.empty()) {
+    props.push_back({"Children", children});
+  }
+
+  SelectionSet insts;
+  for (auto* inst : region->getRegionInsts()) {
+    insts.insert(gui->makeSelected(inst));
+  }
+  props.push_back({"Instances", insts});
+
+  return props;
+}
+
+Selected DbRegionDescriptor::makeSelected(std::any object,
+                                         void* additional_data) const
+{
+  if (auto region = std::any_cast<odb::dbRegion*>(&object)) {
+    return Selected(*region, this, additional_data);
+  }
+  return Selected();
+}
+
+bool DbRegionDescriptor::lessThan(std::any l, std::any r) const
+{
+  auto l_layer = std::any_cast<odb::dbRegion*>(l);
+  auto r_layer = std::any_cast<odb::dbRegion*>(r);
+  return l_layer->getId() < r_layer->getId();
+}
+
+bool DbRegionDescriptor::getAllObjects(SelectionSet& objects) const
+{
+  auto* chip = db_->getChip();
+  if (chip == nullptr) {
+    return false;
+  }
+  auto* block = chip->getBlock();
+  if (block == nullptr) {
+    return false;
+  }
+
+  for (auto* region : block->getRegions()) {
+    objects.insert(makeSelected(region, nullptr));
+  }
+  return true;
+}
+
+//////////////////////////////////////////////////
+
+DbModuleDescriptor::DbModuleDescriptor(odb::dbDatabase* db) :
+    db_(db)
+{
+}
+
+std::string DbModuleDescriptor::getShortName(std::any object) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+  return module->getName();
+}
+
+std::string DbModuleDescriptor::getName(std::any object) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+  return module->getHierarchicalName();
+}
+
+std::string DbModuleDescriptor::getTypeName() const
+{
+  return "Module";
+}
+
+bool DbModuleDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+  auto insts = module->getInsts();
+  if (insts.empty()) {
+    return false;
+  }
+  bbox.mergeInit();
+  for (auto* inst : insts) {
+    auto* box = inst->getBBox();
+    odb::Rect box_rect;
+    box->getBox(box_rect);
+    bbox.merge(box_rect);
+  }
+  return true;
+}
+
+void DbModuleDescriptor::highlight(std::any object,
+                                  Painter& painter,
+                                  void* additional_data) const
+{
+  auto* module = std::any_cast<odb::dbModule*>(object);
+
+  auto* inst_descriptor = Gui::get()->getDescriptor<odb::dbInst*>();
+  for (auto* inst : module->getInsts()) {
+    inst_descriptor->highlight(inst, painter, nullptr);
+  }
+
+  for (auto* children : module->getChildren()) {
+    highlight(children->getMaster(), painter, nullptr);
+  }
```

Comment:
This introduces a full recursion so all instances are highlighted in a single color making it confusing as to what is local and what is not.  Either we should remove this or introduce a way to highlight the instances in another color (or color per child to see grouping).

Reply:
I modified the alpha of the pen for each sublevel in an attempt to indicate the depth of the sub-modules. In general, I liked being able to see all the submodules as well, but if it isn't right, I'm happy to pull it out since that should be accounted for in the design browser later.

Reply:
I find the alpha not very visible, especially when zoomed out.  However the real solution is the hierarchy browser so I'll live with this for now.

Reply:
I agree, once the browser is in we can revisit